eng/restore-toolset.sh

```diff
@@ -7,7 +7,7 @@ function InitializeCustomSDKToolset {
 
   # The following frameworks and tools are used only for testing.
   # Do not attempt to install them in source build.
-  if [[ $properties == *"ArcadeBuildFromSource=true"* ]]; then
+  if [[ $properties == *"ArcadeBuildFromSource=true"* || $product_build == true || $properties == *"DotNetBuildRepo=true"* ]]; then
```

Comment:
Why check both?

Reply:
Right now we don't activate the product build by passing -productBuild in all cases. In the VMR we pass DotNetBuildRepo=true. I wanted to be more resilient. It may be that -productBuild to the build scripts is what we want to use as a standard, but this will require some additional work in some repos that don't use the common scripts right now.